src/move/econia/sources/market.move

```diff
@@ -4196,10 +4216,22 @@ module econia::market {
             limit_price,
             order_id: market_order_id
         };
+        // Check if violated limit price.
+        let violated_limit_price = if (!liquidity_gone) {
```

Comment:
Why check if the limit price is too high or low for the order to be filled after attempting to match? Why not before?

Reply:
@bogberry `violated_limit_price` is only checked after attempted matching because orders that match, incur slippage, and then cancel due to a limit price violation only violate the limit price after matching, not before. Forr this case, if the check were performed before attempted matching then the violated limit price cancel reason would be missed.

Orders that violate the limit price before attempted matching result in `match()` halting without matching, then trip the `violated_limit_price` check here.

So by checking after attempted matching the result is the same for both cases, a cancel event with `CANCEL_REASON_VIOLATED_LIMIT_PRICE`

Reply:
This strikes me as a very convoluted way of performing this check. During the matching process you know when you’ve reached a price level that falls above or below the limit price, and there is still remaining size on the orderbook to fill. 

Attempting to fill the order and mutating the orderbook, borrowing the orderbook _again_ to get the top of the orderbook you were at a moment ago to check against the limit price of the order adds unnecessary steps to perform what should be a very simple check.

Alternatively you could check against the orderbook without performing any mutations first to see if the entire order is fillable. This way you can have just a `borrow_global` at the initial step and abort without ever calling `borrow_global_mut` if the order is impossible to fill.

Reply:
> During the matching process you know when you’ve reached a price level that falls above or below the limit price, and there is still remaining size on the orderbook to fill

@bogberry indeed, see 6a1deeb

Reply:
Why is there not a similar cancel reason emitted for fill-or-kill limit orders? Seems like there's a check missing here.

Reply:
There are no "fill-or-kill" orders but there is a `FILL_OR_ABORT` restriction:

Unlike an immediate-or-cancel order, for example, which emits a `PlaceLimitOrderEvent`, optionally `FillEvent`(s), and optionally a `CancelOrderEvent`, if a fill-or-abort order does not fill completely across the spread the entire transaction aborts because of the complexity associated with rolling back fills. Hence “abort”

A CEX has a cheap database and can rollback memory and simulate fills etc., but for the matching engine it would be difficult to do this and cost more gas

Hence it tries filling and if at the end it doesn’t meet the condition it blows up the transaction with an abort, which means that in terms of events emitted, state changes, it’s like the transaction was never even placed